PopPUNK/network.py

```diff
@@ -453,7 +512,15 @@ def printClusters(G, outPrefix = "_clusters.csv", oldClusterFile = None,
     if oldClusterFile == None and printRef == False:
         raise RuntimeError("Trying to print query clusters with no query sequences")
 
-    newClusters = sorted(nx.connected_components(G), key=len, reverse=True)
+    # get a sorted list of component assignments
+    component_assignments, component_frequencies = gt.label_components(G)
+    component_frequency_ranks = len(component_frequencies) - rankdata(component_frequencies, method = 'ordinal').astype(int)
+    newClusters = [set() for rank in range(len(component_frequency_ranks))]
+    for isolate_index, isolate_name in enumerate(rlist):
+        component = component_assignments.a[isolate_index]
+        component_rank = component_frequency_ranks[component]
+        newClusters[component_rank].add(isolate_name)
```

Comment:
This feels familiar (but I might be wrong) - was the same code also somewhere above? If so, make a function

Reply:
Not sure where there is any substantial repetition. Did some tidying of this file anyway in abf0bc3.